src/nautilus-window.c

```diff
@@ -527,9 +543,36 @@ const GActionEntry win_entries[] = {
 	{ "prompt-root-location", action_prompt_for_location_root },
 	{ "prompt-home-location", action_prompt_for_location_home },
 	{ "go-to-tab", NULL, "i", "0", action_go_to_tab },
+	{ "custom-key-press", NULL, "s", "''", action_custom_key_press },
 };
 
 static void
+nautilus_load_custom_shortcuts(GApplication *app){ 
```

Comment:
Following the suggestion above, I recommend you instead have a function like `nautilus_get_custom_accel_for_script()` instead of this. The function would load the accelerator map and cache it in e.g. a hash-table like structure.